src/simulations/record.py

```diff
@@ -188,6 +186,42 @@ def last_epoch(self, evaluation, method, record_output, batch, done):
             if done and method == "baseline":
                 self.last[self.repeat]["batch"] = batch
 
+    def timer_stats(self, prm, timer_pp, timer_comparison,
+                    timer_optimisation, timer_feasible_data, i):
+        """
+        Calculates the mean, standard deviation and count of
+        the timer used to evaluate the computational burden
+        of some functions and methods.
+        """
+        conditions = [
+            (prm["grd"]["manage_voltage"] or prm["grd"]["manage_agg_power"], timer_pp),
+            (prm["grd"]["manage_voltage"] and prm["grd"]["compare_pandapower_optimisation"],
+                timer_comparison),
+            (True, timer_optimisation),
+            (True, timer_feasible_data),
+        ]
+
+        timer_types = [
+            'timer_pp',
+            'timer_comparison',
+            'timer_opti',
+            'timer_feasible_data'
+        ]
+
+        for condition, timer in conditions:
+            if condition and len(timer) != 0:
+                timer_mean = np.mean(timer)
+                timer_std = np.std(timer)
+                timer_count = len(timer)
+            else:
+                timer_mean = 0
+                timer_std = 0
+                timer_count = 0
+
+        setattr(self, timer_types.pop(0) + '_mean' + '[{}]'.format(i), timer_mean)
+        setattr(self, timer_types.pop(0) + '_std' + '[{}]'.format(i), timer_std)
+        setattr(self, timer_types.pop(0) + '_count' + '[{}]'.format(i), timer_count)
```

Comment:
why add `I`?
also, you can use the `f"xxx{xxx}xxx"` f string format for conciseness.